qa/tasks/cephfs/test_nfs.py

```diff
@@ -534,3 +546,52 @@ def test_cluster_reset_user_config_with_non_existing_clusterid(self):
             # Command should fail for test to pass
             if e.exitstatus != errno.ENOENT:
                 raise
+
+    def test_update_export(self):
+        '''
+        Test update of exports
+        '''
+        self._create_default_export()
+        port, ip = self._get_port_ip_info()
+        self._test_mnt(self.pseudo_path, port, ip)
+        export_block = self._get_export()
+        new_pseudo_path = '/testing'
+        export_block['pseudo'] = new_pseudo_path
+        export_block['access_type'] = 'RO'
+        self.ctx.cluster.run(args=['sudo', 'ceph', 'nfs', 'export', 'update', '-i', '-'],
+                stdin=json.dumps(export_block))
+        self._check_nfs_cluster_status('running', 'NFS Ganesha cluster restart failed')
+        self._write_to_read_only_export(new_pseudo_path, port, ip)
+        self._test_delete_cluster()
+
+    def test_update_export_with_invalid_values(self):
+        '''
+        Test update of export with invalid values
+        '''
+        def update_with_invalid_values(key, value, fsal=False):
+            export_block_new = dict(export_block)
+            if fsal:
+                export_block_new['fsal'] = dict(export_block['fsal'])
+                export_block_new['fsal'][key] = value
+            else:
+                export_block_new[key] = value
+            try:
+                self.ctx.cluster.run(args=['sudo', 'ceph', 'nfs', 'export', 'update', '-i', '-'],
+                        stdin=json.dumps(export_block_new))
+            except CommandFailedError:
+                pass
+
+        self._create_default_export()
+        export_block = self._get_export()
```

Comment:
put these two lines above the definition for `update_with_invalid_values`, otherwise the scoping is weird for `export_block`.

Reply:
Done